src/main/java/nextstep/subway/applicaion/FavoriteService.java

```diff
@@ -0,0 +1,53 @@
+package nextstep.subway.applicaion;
+
+import nextstep.member.application.dto.LoginUser;
+import nextstep.subway.applicaion.dto.FavoriteRequest;
+import nextstep.subway.applicaion.dto.FavoriteResponse;
+import nextstep.subway.domain.Favorite;
+import nextstep.subway.domain.FavoriteRepository;
+import nextstep.subway.domain.Station;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class FavoriteService {
+
+    private final FavoriteRepository favoriteRepository;
+    private final StationService stationService;
+    private final PathService pathService;
+
+    public FavoriteService(FavoriteRepository favoriteRepository, StationService stationService, PathService pathService) {
+        this.favoriteRepository = favoriteRepository;
+        this.stationService = stationService;
+        this.pathService = pathService;
+    }
+
+    @Transactional
+    public FavoriteResponse create(FavoriteRequest request, Long userId) {
+        if(pathService.isConnected(request.getSource(), request.getTarget())){
+            throw new IllegalArgumentException("두 역은 서로 연결되지 않았습니다.");
+        }
+
+        Station source = stationService.findById(request.getSource());
+        Station target = stationService.findById(request.getTarget());
+
+        Favorite favorite = new Favorite(null, userId, source, target);
+        favorite = favoriteRepository.save(favorite);
+
+        return FavoriteResponse.from(favorite);
+    }
+
+    public List<FavoriteResponse> showAll(Long userId) {
+        List<Favorite> favorites = favoriteRepository.findAllByMemberId(userId);
+
+        return favorites.stream().map(ele -> FavoriteResponse.from(ele)).collect(Collectors.toList());
```

Comment:
줄바꿈을 적절하게 사용한다면 동료들이 봤을 때 더 가독성 있는 코드를 만들 수 있어요.
```suggestion
        return favorites.stream()
                .map(it -> FavoriteResponse.from(it))
                .collect(Collectors.toList());
```